derive-encode/src/lib.rs

```diff
@@ -88,22 +88,74 @@ pub fn derive_encode_label_set(input: TokenStream) -> TokenStream {
 }
 
 /// Derive `prometheus_client::encoding::EncodeLabelValue`.
-#[proc_macro_derive(EncodeLabelValue)]
+///
+/// This macro only applies to `enum`s and will panic if you attempt to use it on structs.
+///
+/// At the enum level you can use `#[prometheus(value_case = "lower")]` or `"upper"` to set the
+/// default case of the enum variants.
+///
+/// ```rust
+/// # use prometheus_client::encoding::EncodeLabelValue;
+/// #[derive(Clone, Hash, PartialEq, Eq, EncodeLabelValue, Debug)]
+/// #[prometheus(value_case = "upper")]
+/// enum Method {
+///     Get,
+///     Put,
+/// }
+/// ```
+///
+/// Will encode to label values "GET" and "PUT" in prometheus metrics.
+///
+/// For variants you can use `#[prometheus(lower)]` or `#[prometheus(upper)]` to set the case for
+/// only that variant.
```

Comment:
Do you have a concrete use-case where one would upper or lower case a specific variant only?

Reply:
I do not

it might be better to remove this and wait for a motivated individual to plumb serde in to label generation which could provide all manner of derive customization

Reply:
Do you want me to remove per-variant case changing?

Reply:
Yes please remove. Thank you.